pretix_attestation_plugin/generator/java_generator_wrapper.py

```diff
@@ -0,0 +1,43 @@
+from pretix.base.models import OrderPosition
+
+# Here key indicates .pem file in RFC 5915 format. Before using this generator, key needs
+# to be uploaded through form.
+# Currently, uploaded keys are stored in KEYFILE_DIR variable specified in views.py. Now
+# the directory is "pretix_attestation_plugin/static/pretix_attestation_plugin/keyfiles"
+
+
+def generate_link(order_position: OrderPosition,
+                  path_to_key: str,
+                  path_to_generator: str = 'pretix_attestation_plugin/generator/attestation-all.jar',
+                  ticket_staus: str = '1') -> str:
+    from subprocess import (
+        Popen,
+        PIPE
+    )
+
+    import os.path
+
+    if not os.path.isfile(path_to_key):
+        raise ValueError(f'Key file not found in {path_to_key}')
+
+    if not os.path.isfile(path_to_generator):
+        raise ValueError(f'Generator file not found in {path_to_generator}')
+
+    email = order_position.attendee_email
+    event_id = str(order_position.order.event.id)
+    ticket_id = str(order_position.item.id)
+
+    process = Popen(['java', '-cp', path_to_generator,
```

Comment:
do you get valid URLs this way? Seems the a tool wants numbers for event_id and ticket_id - I think should be strings on their side to  preserve data - will raise with @alphawallet team

Reply:
Not exactly sure how to check the validity of URLs, this generates URLs that start with `?ticket=` followed by some sequence of characters. 

About the types, Popen expects to have arguments as a list of strings or bytes. Internally it is the same as calling the script from your terminal - everything is a sequence of bytes and it is up to the script to parse it properly.

Reply:
Yea - in this case it is visually invalid as it does not yield a URL at all but a error message telling me it wants numbers ;-)

Reply:
Sorry, I didn't get it. What do you mean it is visually invalid? Also, could you explain what you did to get an error message?

Reply:
```
✗ ~/tmp java -cp attestation-all.jar org.devcon.ticket.Issuer key.pem nobody@test.de 123 123 123                             12:59:36
https://ticket.devcon.org/?ticket=MIGYMAkCAXsCAXsCAXsEQQQLFwG-BY_eYpN3lXUEQrQUw6SN9T-_fI81KyET3CrQ5B8mxWFvgqCNC0yV99AH1FafXXdc6GuUEdDcEscVyxrrA0gAMEUCIQDwj8fo4UfNakN63tUPECmo9w1xtLCn5AQR3y-gz0CuRQIgRoksfiLpNmov5vPGPieZO2G5kN3ju7_Ts4QYnpqMgQQ=&secret=8866979750048963284091757617258263722376938844587063505286301348808812438942⏎                                                                        ✓ ~/tmp java -cp attestation-all.jar org.devcon.ticket.Issuer key.pem nobody@test.de 8ABCFD 123 123                          12:59:42
Exception in thread "main" java.lang.NumberFormatException: For input string: "8ABCFD"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:68)
	at java.base/java.lang.Integer.parseInt(Integer.java:652)
	at java.base/java.lang.Integer.parseInt(Integer.java:770)
	at org.devcon.ticket.Issuer.main(Issuer.java:40)
```

Reply:
But the event_id is always numeric, so you won't have anything like 8ABCFD.

Reply:
Sorry for the delay here - there where some discussions. In the end we decided to change event_id to string (the slug of the even) - but it takes a bit longer upstream. So I will merge this now to do some tests - but we will have this one change later on here. But guess on this side it shuld be easy right?

Reply:
No worries. Yeah, I think there shouldn’t be much to change, so should be easy.